model/minsky.cc

```diff
@@ -255,27 +255,47 @@ namespace minsky
 //      throw runtime_error("cannot save to "+fileName);
   }
 
-  void Minsky::paste(float x, float y)
+  void Minsky::paste()
   {
     istringstream is(getClipboard());
     xml_unpack_t unpacker(is);
     schema2::Minsky m;
     xml_unpack(unpacker, "Minsky", m);
     GroupPtr g(new Group);
-    canvas.setItemFocus(model->addGroup(g));
+    model->addGroup(g);
     m.populateGroup(*g);
     // Default pasting no longer occurs as grouped items or as a group within a group. Fix for tickets 1080/1098    
-    g->moveTo(x,y);      
     auto copyOfItems=g->items;
+    size_t count=0;
+    size_t numPaste=0;    
+    if (copyOfItems.size()%2==0) numPaste = copyOfItems.size()/2-g->numWires();
+    else numPaste = copyOfItems.size()/2-g->numWires()-1;    
     for (auto& i: copyOfItems)
-      {
-        model->addItem(i);
-        canvas.selection.toggleItemMembership(i);
-        assert(!i->ioVar());
+      {		
+		 // Ensure only a single copy of a selection of items exists at time 
+		 if (count<numPaste && model->uniqueItems()) 
```

Comment:
I don't follow at all what you're doing with numPaste, an uniqueItems is a quite expensive operation, so shouldn't be done in a loop like this.

Reply:
Thanks, your suggestion of clearing the selection first did the trick.